js/components/ReportSummary.vue

```diff
@@ -33,25 +35,104 @@ export default {
   },
 
   methods: {
+    /**
+     * Emit deleteSummary event.
+     */
     deleteSummary() {
       if (confirm('Permanently delete this summary?')) {
         this.$emit('deleteSummary', this.index);
       }
     },
 
+    /**
+     * Checks if the user has permission to delete a report summary.
+     * @return true if the user has permission to delete a report summary.
+     */
     canDelete() {
       // TODO: implement REDDEV-595
       return true;
+    },
+
+    /**
+     * Checks a value to determine if it is missing. Missing values are those
+     * that are zero in length, undefined, only whitespace, or null.
+     * @return true if the value is missing
+     */
+    missingValue(value) {
+      if (value == null || typeof value === 'undefined') {
+        return true;
+      } else {
+        const stringValue = new String(value);
+        const trimmed = stringValue.trim();
+        return trimmed.length ? false : true;
+      }
     }
```

Comment:
What kinds of values are we checking for here? It looks like this would return false for an empty array or object. Do we care about numeric values? My preference would be to have this more explicitly check for the type to clarify intent. For example:

    else if (typeof value == "string" && !value.trim()) { return true; }

Also, is there a library function that does this?